src/Microsoft.TemplateEngine.Edge/Settings/TemplateInfoReader.cs

```diff
@@ -41,6 +41,7 @@ internal static TemplateInfo FromJObject(JObject entry)
                 }
 
                 info.DefaultName = entry.ToString(nameof(DefaultName));
+                info.PreferDefaultName = entry.ToBool(nameof(PreferDefaultName));
```

Comment:
consider the tests for template cache read/write.
See https://github.com/dotnet/templating/blob/main/test/Microsoft.TemplateEngine.Edge.UnitTests/TemplateCacheTests.cs